framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,554 @@
+              rootMaster.remove(node)
+      # 4. copy slave ET into master ET
+      for node in rootMaster.findall('.//'):
+          if node.tag == 'path':
+              for subNode in node.findall('sequence'):
+                  linkName = subNode.get('name')
+                  if linkName == location:
+                      node.append(rootSlave.find('initial-state').find('fork'))
+                      node.remove(subNode)
+      return copy.deepcopy(rootMaster)
+
+
+  def checkSubBranches(self,root):
+      """
+        This method checks if the provided ET contains sub-branches (i.e., by-pass).
+        This occurs if the node <define-branch> is provided.
+        As an example:
+            <define-branch name="sub-tree1">
+        defines a branch that is linked to multiple ET sequences:
+            <path state="0">
+                <branch name="sub-tree1"/>
+            </path>
+        The scope of this method is to copy the <define-branch> into the sequences of the
+        ET that are linked to <define-branch>
+        @ In,  root, xml.etree.Element, root of the ET
+        @ Out, root, xml.etree.Element, root of the processed ET
+      """
+      eventTree = root.findall('initial-state')
+
+      if len(eventTree) > 1:
+          self.raiseAnError(IOError, 'ETImporter: more than one initial-state identified')
+
+      ### Check for sub-branches
+      subBranches = {}
+      for node in root.findall('define-branch'):
+          subBranches[node.get('name')] = node.find('fork')
+          self.raiseADebug("ETImporter branch identified: " + str(node.get('name')))
+      if len(subBranches) > 0:
+          for node in root.findall('.//'):
+              if node.tag == 'path':
+                  for subNode in node.findall('branch'):
+                      linkName = subNode.get('name')
+                      if linkName in subBranches.keys():
+                          node.append(subBranches[linkName])
+                      else:
+                          self.raiseAnError(RuntimeError, ' ETImporter: branch ' + str(
+                              linkName) + ' linked in the ET is not defined; available branches are: ' + str(
+                              subBranches.keys()))
+
+      for child in root:
+          if child.tag == 'branch':
+              root.remove(child)
+
+      return root
+
+
+  def returnMap(self,outcomes,name):
+      """
+        This method returns a map if the ET contains symbolic sequences.
+        This is needed since since RAVEN requires numeric values for sequences.
+        @ In,  outcomes, list, list that contains all the sequences IDs provided in the ET
+        @ In,  name, string, name of the ET
+        @ Out, map, dict, dictionary containing the map
+      """
+      # check if outputMap contains string ID for  at least one sequence
+      # if outputMap contains all numbers then keep the number ID
+      allFloat = True
+      for seq in outcomes:
+          try:
+              float(seq)
```

Comment:
we have utility functions for achieving this... check mathUtils.py and utils.py

Reply:
done!